subiquity/server/apt.py

```diff
@@ -99,6 +130,67 @@ async def apply_apt_config(self, context):
             self.app, context, 'apt-config', '-t', self.configured_tree.p(),
             config=config_location, private_mounts=True)
 
+    async def run_apt_config_check(self, output: io.StringIO) -> None:
+        """ Run apt-get update (with various options limiting the amount of
+        data donwloaded) in the overlay where the apt configuration was
+        previously deployed. The output of apt-get (stdout + stderr) will be
+        written to the output parameter.
+        Raises a AptConfigCheckError exception if the apt-get command exited
+        with non-zero. """
+        assert self.configured_tree is not None
+
+        pfx = pathlib.Path(self.configured_tree.p())
```

Comment:
So my spike on this topic (or at least one of them) did this checking in an overlay over the configured_tree because running apt with Acquire::IndexTargets::deb::*::DefaultEnabled=false actually deleted the indices for all the types that were disabled, so that the entire e.g. release Packages files got downloaded again during the install. It would be good to check that this isn't happening in your branch, or to switch to doing the check in an overlay (on an overlay!) again if it is.

Reply:
Otherwise this method is very nice and easy to read :-)

Reply:
> So my spike on this topic (or at least one of them) did this checking in an overlay over the configured_tree because running apt with Acquire::IndexTargets::deb::*::DefaultEnabled=false actually deleted the indices for all the types that were disabled, so that the entire e.g. release Packages files got downloaded again during the install. It would be good to check that this isn't happening in your branch, or to switch to doing the check in an overlay (on an overlay!) again if it is.

I think I see what you mean. Currently, I'm calling apply_apt_config each time I want to run a mirror check, therefore a fresh configured_tree gets deployed every time. We might see the issue you describe going forward if we decide to avoid creating new overlays all the time.

Reply:
oh i see, that makes sense too